Eatery/Controllers/Eateries/Campus/ScrollableViewController.swift

```diff
@@ -0,0 +1,374 @@
+        }
+        tabBar = ScrollableTextTabBarControl(sections: titles, padding: 16)
+        tabBar.directionalLayoutMargins = NSDirectionalEdgeInsets(top: 0, leading: 16, bottom: 0, trailing: 16)
+        tabBar.addTarget(self, action: #selector(manualScroll), for: .valueChanged)
+        tabBar.backgroundColor = .white
+        tabBar.layer.zPosition = 2
+
+        tabBar.snp.makeConstraints { make in
+            make.height.equalTo(40)
+        }
+
+        tabBarStack = UIStackView(arrangedSubviews: [headerContainer, tabBar])
+        tabBarStack.axis = .vertical
+    }
+
+    private func createMenuStack() {
+        menuItemsView = UIView()
+        menuItemsView.backgroundColor = .white
+
+        menuItemsStack = UIStackView(arrangedSubviews: originalViews)
+        menuItemsStack.axis = .vertical
+        menuItemsView.addSubview(menuItemsStack)
+        view.addSubview(menuItemsView)
+        view.addSubview(tabBarStack) // Tabbar added here to account for z position
+
+        tabBarStack.snp.makeConstraints { make in
+            make.top.equalToSuperview()
+            make.leading.trailing.equalToSuperview()
+        }
+
+        menuItemsView.snp.makeConstraints { make in
+            make.top.equalTo(tabBar.snp.bottom)
+            make.leading.trailing.bottom.equalToSuperview()
+        }
+
+        menuItemsStack.snp.makeConstraints { make in
+            make.top.bottom.equalToSuperview()
+            make.leading.trailing.equalToSuperview().inset(5)
+        }
+
+    }
+
+    private func createHighToLowStack() {
+        let reversedItems = Array(allItems.reversed())
+        let sortedVC = CampusEateryExpandedMenuViewController(eatery: eatery, category: "", menu: reversedItems)
+
+        menuItemsHighToLowStack = UIStackView(arrangedSubviews: [sortedVC.view])
+        menuItemsHighToLowStack.axis = .vertical
+        menuItemsView.addSubview(menuItemsHighToLowStack)
+        menuItemsHighToLowStack.isHidden = true
+
+        self.menuItemsHighToLowStack.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    private func createLowToHighStack() {
+        let sortedVC = CampusEateryExpandedMenuViewController(eatery: eatery, category: "", menu: allItems)
+
+        menuItemsLowToHighStack = UIStackView(arrangedSubviews: [sortedVC.view])
+        menuItemsLowToHighStack.axis = .vertical
+        menuItemsView.addSubview(menuItemsLowToHighStack)
+        menuItemsLowToHighStack.isHidden = true
+
+        self.menuItemsLowToHighStack.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    @objc func manualScroll() {
+        let currentPos = self.scrollView?.contentOffset.y
+        let viewControllerPosOnParent = self.view.frame.origin.y
+        let viewPos = originalViews[tabBar.selectedSegmentIndex].frame.minY
+        let minOffset: CGFloat = 4
+        let targetPos = viewPos - self.scrollOffset + viewControllerPosOnParent + minOffset
```

Comment:
sometimes you use self and sometimes you don't for the class variables, i'd recommend staying consistent with one style